compiler/ccgtypes.nim

```diff
@@ -220,7 +220,8 @@ proc isInvalidReturnType(conf: ConfigRef; rettype: PType): bool =
   # such a poor programming language.
   # We exclude records with refs too. This enhances efficiency and
   # is necessary for proper code generation of assignments.
-  if rettype == nil or (tfByCopy notin rettype.flags and getSize(conf, rettype) > conf.target.floatSize*3):
+  if rettype == nil or ({tfByCopy, tfCDecl} * rettype.flags == {} and
```

Comment:
Instead of a new flag, pass the proc type too to `isInvalidReturnType` and check for: `procType.callConv in {ccClosure, ccInline, ccDefault, ccNimcall}` -- only for these the transformation is allowed.

Reply:
ok